packages/app/src/app/overmind/internalActions.ts

```diff
@@ -353,3 +352,114 @@ export const closeTabByIndex: Action<number> = ({ state }, tabIndex) => {
 
   state.editor.tabs.splice(tabIndex, 1);
 };
+
+export const onApiError: Action<ApiError> = (
+  { state, actions, effects },
+  error
+) => {
+  const { response } = error;
+
+  if (response.status === 401) {
+    // We need to implement a blocking modal to either sign in or refresh the browser to
+    // continue in an anonymous state
+  }
+
+  if (!response || response.status >= 500) {
+    effects.analytics.logError(error);
+  }
+
+  const result = response.data;
+
+  if (result) {
+    if ('errors' in result) {
+      const errors = values(result.errors)[0];
+      if (Array.isArray(errors)) {
+        if (errors[0]) {
+          error.message = errors[0]; // eslint-disable-line no-param-reassign,prefer-destructuring
+        }
+      } else {
+        error.message = errors; // eslint-disable-line no-param-reassign
+      }
+    } else if (result.error) {
+      error.message = result.error; // eslint-disable-line no-param-reassign
+    } else if (response.status === 413) {
+      error.message = 'File too large, upload limit is 5MB.';
+    }
+  }
+
+  if (error.message.startsWith('You need to sign in to create more than')) {
```

Comment:
Nice!! And in the future we can even move this message to the fork call or something.

One thing that I would like to see in the future is that the processed error is thrown something like:

```
{
  apiError: AxiosError,
  message: string,
  etc...
}
```

Because our API has a way of return multiple kinds of messages based on field based validations or other errors.